src/main/scala/beam/utils/FileUtils.scala

```diff
@@ -337,4 +338,21 @@ object FileUtils extends LazyLogging {
     }
     Await.result(Future.sequence(futures), atMost)
   }
+
+  def gzipFile(file: String, deleteSourceFile: Boolean = false): Unit = {
+    val filePath = Paths.get(file)
+    if (!filePath.toFile.isFile) {
+      throw new IllegalStateException(s"Not a file: `$file`")
+    }
+
+    using(IOUtils.getBufferedWriter(file, true)) { bw =>
```

Comment:
It took me a lot of time to understand how this worked because the true was not indicative of how this turns into a gzip. For future readability I would suggest to use the named argument:

```suggestion
    using(IOUtils.getBufferedWriter(file, useCompression = true)) { bw =>
```